### PR TITLE
networkx 3.3

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Destination channel:** main

### Links

- [PKG-3167](https://anaconda.atlassian.net/browse/PKG-3167)
- release-notes: https://github.com/networkx/networkx/releases
- release-notes-tagged: https://github.com/networkx/networkx/releases/tag/v3.3
- release-diff: https://github.com/networkx/networkx/compare/v3.2.1...v3.3
- license: https://github.com/networkx/networkx/blob/main/LICENSE.txt
- requirements-tagged:
  - https://github.com/networkx/networkx/blob/networkx-3.3/pyproject.toml

### Explanation of changes:

- Skip py<310
- Update run_constrained pinnings
- Remove `networkx.tests` from imports  as redundant

[PKG-3167]: https://anaconda.atlassian.net/browse/PKG-3167?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,13 +22,13 @@ requirements:
   run:
     - python
   run-constrained:
```
Comment (lorepirri) [inline recipe/meta.yaml]:
```suggestion
  run_constrained:
```

**Review by lorepirri — COMMENTED** [review]
LGTM,
tests are in the wheel, could you try `pytest -v --pyargs networkx`?

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Good catch! Thanks!

**Review by ifitchet — COMMENTED** [review]
`pyproject.toml` says `Homepage: https://networkx.org/` which is where `https://networkx.github.io/` redirects to.

**Comment by skupr-anaconda** [comment]
> `pyproject.toml` says `Homepage: https://networkx.org/` which is where `https://networkx.github.io/` redirects to.

@ifitchet I updated it

**Review by lorepirri — APPROVED** [review]
LGTM, thank you for adding tests!

**Review by ifitchet — APPROVED** [review]
LGTM

**Pull request merged**